src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,455 @@
+
+        /// <summary>
+        ///  Returns the pressure in Pa, in Q24.8 format (24 integer bits and 8 fractional bits).
+        ///  Output value of “24674867” represents 24674867/256 = 96386.2 Pa = 963.862 hPa
+        /// </summary>
+        /// <param name="adcPressure">
+        /// The pressure value read from the device
+        /// </param>
+        /// <returns>
+        ///  Pressure in hPa
+        /// </returns>
+        private long CompensatePressure(int adcPressure)
+        {
+            //The pressure is calculated using the compensation formula in the BMP280 datasheet
+            long var1 = _temperatureFine - 128000;
+            long var2 = var1 * var1 * (long)_calibrationData.DigP6;
+            var2 = var2 + ((var1 * (long)_calibrationData.DigP5) << 17);
+            var2 = var2 + ((long)_calibrationData.DigP4 << 35);
+            var1 = ((var1 * var1 * (long)_calibrationData.DigP3) >> 8) + ((var1 * (long)_calibrationData.DigP2) << 12);
+            var1 = (((((long)1 << 47) + var1)) * (long)_calibrationData.DigP1) >> 33;
+            if (var1 == 0)
+            {
+                return 0; //Avoid exception caused by division by zero
+            }
+            //Perform calibration operations
+            long p = 1048576 - adcPressure;
+            p = (((p << 31) - var2) * 3125) / var1;
+            var1 = ((long)_calibrationData.DigP9 * (p >> 13) * (p >> 13)) >> 25;
+            var2 = ((long)_calibrationData.DigP8 * p) >> 19;
+            p = ((p + var1 + var2) >> 8) + ((long)_calibrationData.DigP7 << 4);
+            return p;
+        }
+
+        /// <summary>
+        ///  Reads an 8 bit value from a register
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        private byte Read8FromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                _i2cDevice.WriteByte(register);
+                byte value = _i2cDevice.ReadByte();
+                return value;
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        ///  Reads a 16 bit value over I2C 
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        private ushort Read16FromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                byte[] readBuffer = { 0x00, 0x00 };
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(readBuffer);
+                int h = readBuffer[1] << 8;
+                int l = readBuffer[0];
+                ushort value = (ushort)(h + l);
```

Comment:
Looks like you are doing the Endianess yourself here which might be tricky and not work all of the time. Have you tried using [BinaryPrimitives](https://github.com/dotnet/corefx/blob/master/src/System.Memory/ref/System.Memory.cs#L256) class?

Reply:
Now I have, thanks.